core/src/main/java/io/atomix/core/multimap/impl/ConsistentSetMultimapProxy.java

```diff
@@ -26,180 +26,140 @@
-    return this.<ContainsValue, Boolean>invokeAll(CONTAINS_VALUE, new ContainsValue(value))
-        .thenApply(results -> results.filter(Predicate.isEqual(true)).findFirst().orElse(false));
+    return applyAll(service -> service.containsValue(value))
+        .thenApply(results -> results.anyMatch(Predicate.isEqual(true)));
   }
 
   @Override
   public CompletableFuture<Boolean> containsEntry(String key, byte[] value) {
-    return this.<ContainsEntry, Boolean>invokeAll(CONTAINS_ENTRY, new ContainsEntry(key, value))
-        .thenApply(results -> results.filter(Predicate.isEqual(true)).findFirst().orElse(false));
+    return applyBy(key, service -> service.containsEntry(key, value));
   }
 
   @Override
   public CompletableFuture<Boolean> put(String key, byte[] value) {
-    return invokeBy(key, PUT, new Put(key, Collections.singletonList(value), null));
+    return applyBy(key, service -> service.put(key, value));
   }
 
   @Override
   public CompletableFuture<Boolean> remove(String key, byte[] value) {
-    return invokeBy(key, REMOVE, new MultiRemove(key, Collections.singletonList(value), null));
+    return applyBy(key, service -> service.remove(key, value));
   }
 
   @Override
   public CompletableFuture<Boolean> removeAll(String key, Collection<? extends byte[]> values) {
-    return invokeBy(key, REMOVE, new MultiRemove(key, (Collection<byte[]>) values, null));
+    return applyBy(key, service -> service.removeAll(key, values));
   }
 
   @Override
   public CompletableFuture<Versioned<Collection<? extends byte[]>>> removeAll(String key) {
-    return invokeBy(key, REMOVE_ALL, new RemoveAll(key, null));
+    return applyBy(key, service -> service.removeAll(key));
   }
 
   @Override
   public CompletableFuture<Boolean> putAll(String key, Collection<? extends byte[]> values) {
-    return invokeBy(key, PUT, new Put(key, values, null));
+    return applyBy(key, service -> service.putAll(key, values));
   }
 
   @Override
   public CompletableFuture<Versioned<Collection<? extends byte[]>>> replaceValues(
       String key, Collection<byte[]> values) {
-    return invokeBy(key, REPLACE, new Replace(key, values, null));
+    return applyBy(key, service -> service.replaceValues(key, values));
   }
 
   @Override
   public CompletableFuture<Void> clear() {
-    return invokeAll(CLEAR).thenApply(v -> null);
+    return acceptAll(service -> service.clear());
   }
 
   @Override
   public CompletableFuture<Versioned<Collection<? extends byte[]>>> get(String key) {
-    return invokeBy(key, GET, new Get(key));
+    return applyBy(key, service -> service.get(key));
   }
 
   @Override
   public CompletableFuture<Set<String>> keySet() {
-    return this.<Set<String>>invokeAll(KEY_SET)
-        .thenApply(results -> results.reduce((s1, s2) -> ImmutableSet.<String>builder().addAll(s1).addAll(s2).build()).orElse(ImmutableSet.of()));
+    return applyAll(service -> service.keySet())
+        .thenApply(results -> results.reduce((s1, s2) ->
```

Comment:
wouldn't collect make more sense here?

Reply:
`results` is a `Stream<Set<String>>` so it has to be reduced before it can be collected

Reply:
Gotcha